### PR TITLE
Fix the deploymentTool filtering

### PR DESCRIPTION
**Pull request by inercia**
We are currently filtering out some new `deploymentTool` values. This simplifies the `deploymentTool` filtering...

### CODE REVIEW

File: templates/_helpers.tpl [hunk 1]
```diff
@@ -61,18 +61,9 @@ Define the http port of the Ambassador service
 {{- end -}}
 
 {{/*
-Filter the valid deploymentTool
+Return a valid managed-by. By default "Helm", but can be overriden with `deploymentTool`
 */}}
 {{- define "ambassador.managedBy" -}}
-    {{- if .Values.deploymentTool -}}
```
Comment (khussey) [inline templates/_helpers.tpl:67]: Rather than removing the filters, should we add checks for the additional values we now consider valid?

Comment (inercia) [reply templates/_helpers.tpl:67]: I think it is not worth it.

File: templates/_helpers.tpl [hunk 2]
```diff
@@ -61,18 +61,9 @@ Define the http port of the Ambassador service
...
-            edgectl
-        {{- else if (eq .Values.deploymentTool "amb-oper") -}}
-            amb-oper
-        {{- else -}}
-            {{ .Release.Service -}}
-        {{- end -}}
-    {{- else -}}
-        {{- .Release.Service -}}
-    {{- end -}}
-{{- end -}}
```
Comment (khussey) [inline templates/_helpers.tpl:78]: Isn't this "end" still needed here?

**Comment by nbkrause** [comment]
If we do not think there is value in validating that `deploymentTool` is something we expect then we should just remove the `ambassador.deploymentTool` from the `_helpers.tpl` and use `.Values.deploymentTool` in the templates.

File: templates/_helpers.tpl (hunk 2 above)
Comment (inercia) [reply templates/_helpers.tpl:78]: Yes, it is.

**Comment by inercia** [comment]
> If we do not think there is value in validating that deploymentTool is something we expect then we should just remove the ambassador.deploymentTool from the _helpers.tpl and use .Values.deploymentTool in the templates.

We still want a `Helm` default value. `deploymentTool` will _never_ be provided by users, only by our tools (ie, the Operator or `edgectl install`). So we want to have `Helm` as default value...

**Comment by nbkrause** [comment]
> > If we do not think there is value in validating that deploymentTool is something we expect then we should just remove the ambassador.deploymentTool from the _helpers.tpl and use .Values.deploymentTool in the templates.
> 
> We still want a `Helm` default value. `deploymentTool` will _never_ be provided by users, only by our tools (ie, the Operator or `edgectl install`). So we want to have `Helm` as default value...

Then I would just 
```
{{ if .Values.deploymentTool }}
app.kubernetes.io/managed-by: {{ .Values.deploymentTool }}
{{ else }}
app.kubernetes.io/managed-by: {{ .Release.Service }}
{{ end }}
```
No reason to add a templated value in `_helpers.tpl` that can be solved using control flows IMO

**Comment by inercia** [comment]
> No reason to add a templated value in _helpers.tpl that can be solved using control flows IMO

@nbkrause I have updated the PR with the changes you suggested.

**Pull request merged**